viewer/packages/360-images/src/collection/Image360CollectionFactory.ts

```diff
@@ -74,11 +74,19 @@ export class Image360CollectionFactory<T> {
         );
       });
 
-    const collectionIds = uniq(historicalDescriptors.map(desc => desc.collectionId));
+    const uniqueCollections = uniqBy(historicalDescriptors, desc => desc.collectionId);
+
+    assert(uniqueCollections.length === 1);
 
-    assert(collectionIds.length === 1);
+    const { collectionId, collectionLabel } = uniqueCollections[0];
 
```

Comment:
Is this necessary? :sweat_smile: Aren't we already pretty sure the `collectionId` is the same for all descriptors? Can we just use `historicalDescriptors[0]`?

Reply:
Nevermind, I missed the previous code :sweat: I agree this sanity check should be somewhere